File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
import sys

N, M, V = map(int, sys.stdin.readline().split())

graph = [[0] * (N+1) for _ in range(N+1)]
visited = [0] * (N+1)

def dfs(V):
    visited[V] = 1

    print(V, end=" ")

    for i in range(1, N+1):
        if(graph[V][i] == 1 and visited[i] == 0):
            dfs(i)

def bfs(V):
    queue = [V]
    visited[V] = 0

    while queue:
        V = queue.pop(0)
        print(V, end=" ")

        for i in range(1, N+1):
            if(graph[V][i] == 1 and visited[i] == 1):
                queue.append(i)
                visited[i] = 0

for _ in range(M):
    start, end = map(int, sys.stdin.readline().split())
    graph[start][end] = graph[end][start] = 1

dfs(V)
print()
bfs(V)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main {
  public static int nodes;
  public static int edgeNum;
  public static int start;
  public static int[][] edges;
  public static boolean[] visited;

  public static void main(String[] args) throws IOException {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    StringTokenizer st = new StringTokenizer(br.readLine());

    nodes = Integer.parseInt(st.nextToken());
    edgeNum = Integer.parseInt(st.nextToken());
    start = Integer.parseInt(st.nextToken());
    edges = new int[nodes + 1][nodes + 1];
    visited = new boolean[nodes + 1];

    for (int i = 0; i < edgeNum; i++) {
      st = new StringTokenizer(br.readLine());
      int startEdge = Integer.parseInt(st.nextToken());
      int endEdge = Integer.parseInt(st.nextToken());

      edges[startEdge][endEdge] = 1;
      edges[endEdge][startEdge] = 1;
    }

    DFS(start);
    System.out.println();

    visited = new boolean[nodes + 1];
    BFS(start);
  }

  public static void DFS(int start) {
    visited[start] = true;
    System.out.print(start + " ");
    for (int i = 1; i <= nodes; i++) {
      if (edges[start][i] == 1 && !visited[i]) {
        DFS(i);
      }
    }
  }

  public static void BFS(int start) {
    Queue<Integer> queue = new LinkedList<>();
    queue.add(start);
    visited[start] = true;
    System.out.print(start + " ");

    while (!queue.isEmpty()) {
      int cur = queue.poll();
      for (int i = 1; i <= nodes; i++) {
        if (edges[cur][i] == 1 && !visited[i]) {
          queue.add(i);
          visited[i] = true;
          System.out.print(i + " ");
        }
      }
    }
  }
}